portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,17 +80,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String newComment = request.getParameter("comment-input");
     long timestamp = System.currentTimeMillis();
 
+    // Get current user email
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+
     if (newComment != null){
+      // calculate sentiment
+      Document doc = Document.newBuilder().setContent(newComment).setType(Document.Type.PLAIN_TEXT).build();
```

Comment:
I would wrap this logic into its own function taking in a string (newComment) and outputting a float (sentimentScore).